### PR TITLE
Add a better error message when decoding a bare `None`

### LINKED ISSUE #896: dhall-to-json None value not supported for optionals.
```
 $ dhall-to-json <<< "Some 1"
1
$  dhall-to-json <<< "None"
Error: Cannot translate to JSON
```

### PR DESCRIPTION
**Pull request by Gabriel439**
Fixes #896

### CODE REVIEW

File: dhall-json/src/Dhall/JSON.hs [hunk 1]
```diff
@@ -202,19 +200,43 @@ import qualified Options.Applicative
...
 -}
-data CompileError = Unsupported (Expr X X) deriving (Typeable)
+data CompileError
+    = Unsupported (Expr X X)
+    | BareNone
 
 instance Show CompileError where
+    show BareNone =
+       Data.Text.unpack $
+            _ERROR <> ": A bare ❰None❱ is not valid                                          \n\
```
Comment (Profpatsch) [inline dhall-json/src/Dhall/JSON.hs]: `❰None❱ on its own is not valid`

File: dhall-json/src/Dhall/JSON.hs [hunk 2]
```diff
@@ -202,19 +200,43 @@ import qualified Options.Applicative
...
+            \                                                                                \n\
+            \Explanation: The conversion to JSON/YAML does not accept ❰None❱ in isolation as \n\
+            \a valid way to represent ❰null❱.  In Dhall, ❰None❱ is a function whose input is \n\
+            \a type and whose output is an absent value of that type.                        \n\
```
Comment (Profpatsch) [inline dhall-json/src/Dhall/JSON.hs]: The “output is an absent value of that type” formulation is a bit confusing.

```whose output is an `Optional` of that type```

File: dhall-json/src/Dhall/JSON.hs [hunk 3]
```diff
@@ -202,19 +200,43 @@ import qualified Options.Applicative
...
+            \                                                                                \n\
+            \For example:                                                                    \n\
+            \                                                                                \n\
+            \                                                                                \n\
+            \    ┌─────────────────────────────────┐  This is a function whose result is an  \n\
```
Comment (Profpatsch) [inline dhall-json/src/Dhall/JSON.hs]: ``` `None` is a function```

File: dhall-json/src/Dhall/JSON.hs [hunk 4]
```diff
@@ -202,19 +200,43 @@ import qualified Options.Applicative
...
+            \    │ None : ∀(a : Type) → Optional a │  ❰Optional❱ value, but the function     \n\
+            \    └─────────────────────────────────┘  itself is not a valid ❰Optional❱ value \n\
+            \                                                                                \n\
+            \                                                                                \n\
+            \    ┌─────────────────────────────────┐                                         \n\
+            \    │ None Natural : Optional Natural │  This is a valid ❰Optional❱ value (an   \n\
```
Comment (Profpatsch) [inline dhall-json/src/Dhall/JSON.hs]: ``` `None Natural` is a valid```

File: dhall-json/src/Dhall/JSON.hs [hunk 5]
```diff
@@ -202,19 +200,43 @@ import qualified Options.Applicative
...
+            \    └─────────────────────────────────┘  absent ❰Natural❱ number in this case)  \n\
+            \                                                                                \n\
+            \                                                                                \n\
+            \The conversion to JSON/YAML only translates the latter form to ❰null❱.          "
```
Comment (Profpatsch) [inline dhall-json/src/Dhall/JSON.hs]: ```only translates the fully applied form (e.g. `None Natural`)```

**Comment by Profpatsch** [comment]
I have a few suggestions that make the error message a bit clearer.

**Comment by Gabriel439** [comment]
@Profpatsch: Since this is already merged, could you submit a pull request?

**Pull request merged**